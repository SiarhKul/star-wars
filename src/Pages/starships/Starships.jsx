import React from "react";
import { useSelector } from "react-redux";
import { ToastContainer } from "react-toastify";
import { ButtonLoadMore } from "../../components";
import { bodyCardStarships } from "../../components/bodyCards/bodyCardStarships";
import { Cards } from "../../components/card/Cards";
import Popup from "../../components/popup/Popup";
import { GET_MORE_STARSHIPS } from "../../redux/actions/actions";
import { history } from "../../redux/reducers";
import { StarshipsFragmentPopup } from "./StarshipsFragmentPopup";

export const Starships = () => {
	const contenCards = useSelector(state => state.dataFromServer.starships);
	const isLoadedStarships = useSelector(
		state => state.loading.isLoadedStarships
	);
	const { isVisiblePopup } = useSelector(state => state.loading);
	const { pathname } = history.location;

	return (
		<main className="main-people">
			<Cards
				contenCards={contenCards}
				path={pathname}
				name="name"
				BodyComponent={bodyCardStarships}
			/>
			{!isLoadedStarships && <ButtonLoadMore action={GET_MORE_STARSHIPS} />}
			{isVisiblePopup && <Popup FragmentPopup={StarshipsFragmentPopup} />}
			<ToastContainer />
		</main>
	);
};
