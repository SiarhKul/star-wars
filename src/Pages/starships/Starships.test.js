import React from "react";
import { Provider } from "react-redux";
import { Starships } from "./Starships";
import configureStore from "redux-mock-store";

const mockStore = configureStore();

describe("Test <Starships/>", () => {
	let component;
	let store;

	const getMockInitialStore = showButton => {
		return {
			dataFromServer: {
				starships: [],
			},
			loading: {
				isLoadedStarships: showButton,
			},
			router: {
				location: {
					pathname: "/starships",
				},
			},
		};
	};

	const starhips = () =>
		mount(
			<Provider store={store}>
				<Starships />
			</Provider>
		);

	beforeEach(() => {
		store = mockStore(getMockInitialStore(true));
		component = starhips();
	});

	it("render snapshot <Starships/> ", () => {
		expect(component).toMatchSnapshot();
	});

	it("should render ButtonLoadMore in <Starships/> if  all data has been loaded", () => {
		expect(component.find("ButtonLoadMore").exists()).toBe(false);
	});

	it("should render ButtonLoadMore in <Starships/> if  all data has not been loaded", () => {
		store = mockStore(getMockInitialStore(false));
		component = starhips();
		expect(component.find("ButtonLoadMore").exists()).toBe(true);
	});

	it("should render alert message in  <Starships/>", () => {
		expect(component.find("ToastContainer")).toHaveLength(1);
	});

	it("should render  <Starships/>", () => {
		expect(component.find("Cards")).toHaveLength(1);
	});
});
