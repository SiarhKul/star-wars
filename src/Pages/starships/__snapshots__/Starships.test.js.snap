// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Starships/> render snapshot <Starships/>  1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Starships>
    <main
      className="main-people"
    >
      <Loader>
        <div
          className="spinner"
        />
      </Loader>
      <ToastContainer
        autoClose={5000}
        closeButton={[Function]}
        closeOnClick={true}
        draggable={true}
        draggableDirection="x"
        draggablePercent={80}
        hideProgressBar={false}
        newestOnTop={false}
        pauseOnFocusLoss={true}
        pauseOnHover={true}
        position="top-right"
        role="alert"
        rtl={false}
        transition={[Function]}
      >
        <div
          className="Toastify"
        />
      </ToastContainer>
      <Cards
        BodyComponent={[Function]}
        contenCards={Array []}
        name="name"
        path="/"
      >
        <div
          className="cards-container"
        />
      </Cards>
    </main>
  </Starships>
</Provider>
`;
